### PR TITLE
feat: display enum variants

### LINKED ISSUE #12: Display enum variants.
Currently the docs only show "type string" and don't show the enum variants for enums. This is annoying. The enum variants are in the CRD, so the info is there.

It's not clear how to display the variants yet.

### PR DESCRIPTION
**Pull request by fhennig**
fixes #12

### CODE REVIEW

**Comment by fhennig** [comment]
see example here: https://deploy-preview-23--stackabletech-crd-docs.netlify.app/23.11.0/nifi.stackable.tech/nificluster/v1alpha1/#spec-clusterConfig-sensitiveProperties-algorithm

File: js/doc.js [hunk 1]
```diff
@@ -113,13 +114,18 @@ function SchemaPart({ key, property, parent, parentSlug }) {
         type = `[]object`;
       }
     }
+    if (property.Enum) {
+      type = type + ': enum';
```
Comment (Techassi) [inline js/doc.js:118]: We should probably use template strings:

```suggestion
      type = `${type}: enum`
```

File: js/doc.js [hunk 2]
```diff
@@ -98,11 +98,12 @@ function CRD() {
 }
 
 function SchemaPart({ key, property, parent, parentSlug }) {
-  const [props, propKeys, required, type, schema] = useMemo(() => {
+  const [props, propKeys, required, type, schema, enumvals] = useMemo(() => {
     let schema = property;
     let props = property.Properties || {};
 
     let type = property.Type;
+    let enumvals = null;
```
Comment (Techassi) [inline js/doc.js:106]: Why not return an empty array instead of null, when the property is not an enum? See also below.

File: js/doc.js [hunk 3]
```diff
@@ -113,13 +114,18 @@ function SchemaPart({ key, property, parent, parentSlug }) {
         type = `[]object`;
       }
     }
+    if (property.Enum) {
+      type = type + ': enum';
+      enumvals = property.Enum;
```
Comment (Techassi) [inline js/doc.js:119]: Returning the enum variants or an empty array:

```suggestion
      enumvals = property.Enum ?? [];
```

**Review by Techassi — CHANGES_REQUESTED** [review]

File: js/doc.js (hunk 3 above)
Comment (Techassi) [reply js/doc.js:119]: We can use this in line 166.

Comment (fhennig) [reply js/doc.js:119]: Ahh, now I get it

**Review by Techassi — APPROVED** [review]

**Pull request merged**